tests/ballerina-unit-test/src/test/resources/test-src/jdbc/transaction/sql_transaction_test.bal

```diff
@@ -782,6 +782,41 @@ function testLocalTransactonWithSelect() returns (int, int) {
     return (returnVal, count);
 }
 
+function testLocalTransactionWithSelectAndHasNextIteration() returns (int, int) {
+    endpoint jdbc:Client  testDB {
+        url: "jdbc:hsqldb:file:./target/tempdb/TEST_SQL_CONNECTOR_TR",
+        username: "SA",
+        poolOptions: { maximumPoolSize: 1 }
+    };
+
+    _ = testDB->update("Insert into Customers (firstName,lastName,registrationID,creditLimit,country)
+                                values ('James', 'Clerk', 901, 5000.75, 'USA')");
+    _ = testDB->update("Insert into Customers (firstName,lastName,registrationID,creditLimit,country)
+                                values ('James', 'Clerk', 901, 5000.75, 'USA')");
+
+    int returnVal = 0;
+    int count;
+    transaction {
+        table<ResultCount> dt1 = check testDB->select("Select COUNT(*) as countval from Customers where
+            registrationID = 901", ResultCount);
+        while (dt1.hasNext()) {
+            ResultCount rs = check <ResultCount>dt1.getNext();
+            count = rs.COUNTVAL;
+        }
+
+        table<ResultCount> dt2 = check testDB->select("Select COUNT(*) as countval from Customers where
+            registrationID = 901", ResultCount);
+        while (dt2.hasNext()) {
+            ResultCount rs = check <ResultCount>dt2.getNext();
+            count = rs.COUNTVAL;
+        }
+    } onretry {
+        returnVal = -1;
+    }
+    testDB.stop();
```

Comment:
@Manuri AFAIU Inside a transaction we do not close the connection it self. This is because same connection will be used for each sql query. If that the case, Is  testDB.stop() will close the connection? 
What happens if retry called, then testDB.stop() will not get executed.

Reply:
testDB.stop() will shutdown the complete connection pool. However, before coming to that point, at the end of the transaction scope, the connection which was used for that transaction will be committed and closed.